packages/block-library/src/form/edit.js

```diff
@@ -51,6 +56,17 @@ const TEMPLATE = [
 ];
 
 const Edit = ( { attributes, setAttributes, clientId } ) => {
+	const defaultSubmissionMethod = 'email';
+
+	const resetAllSettings = () => {
+		setAttributes( {
+			submissionMethod: defaultSubmissionMethod,
+			email: '',
+			action: '',
+			method: 'post',
```

Comment:
```suggestion
			submissionMethod: 'email',
			email: undefined,
			action: undefined,
			method: 'post',
```

- Personally, I don't think it's necessary to make the default value of the `submissionMethod` attribute a constant.
- The default value for the `email` and `action` attributes is `undefined`, not an empty string. Otherwise, unnecessary empty strings would be saved as block comments when resetting.